x/launch/keeper/msg_server_create_chain.go

```diff
@@ -0,0 +1,65 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/tendermint/spn/x/launch/types"
+	profiletypes "github.com/tendermint/spn/x/profile/types"
+)
+
+func (k msgServer) CreateChain(goCtx context.Context, msg *types.MsgCreateChain) (*types.MsgCreateChainResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Get the coordinator ID associated to the sender address
+	coordID, found := k.profileKeeper.CoordinatorIdFromAddress(ctx, msg.Coordinator)
+	if !found {
+		return nil, sdkerrors.Wrap(profiletypes.ErrCoordAddressNotFound, msg.Coordinator)
+	}
+
+	// Compute the chain id
+	chainNameCount, found := k.GetChainNameCount(ctx, msg.ChainName)
+	if !found {
+		chainNameCount = types.ChainNameCount{
+			ChainName: msg.ChainName,
+			Count:     0,
+		}
+	}
+	chainID := types.ChainIDFromChainName(msg.ChainName, chainNameCount.Count)
+	chainNameCount.Count++
+
+	// chainID must always be unique by design
+	// if it already exists then something is wrong in the protocol
+	_, found = k.GetChain(ctx, chainID)
+	if found {
+		panic(fmt.Sprintf("chain id %s already exists", chainID))
```

Comment:
I think it makes sense to use the critical error here as well

Reply:
Yep, wasn't merged yet